### PR TITLE
Pharmacy selection filters

### PR DESCRIPTION
**Pull request by jknapp25**
Refetch pharmacies with filter criteria when filter enabled. Also some copy updates, pluralization, cleanup.

https://github.com/Photon-Health/client/assets/3934326/170b30c1-d08b-4f2f-a231-dddfa6752dad

### CODE REVIEW

**Review by photonsFardad — APPROVED** [review]

File: apps/patient/src/components/PharmacyCard.tsx [hunk 1]
```diff
@@ -119,6 +121,7 @@ export const PharmacyCard = memo(function PharmacyCard({
       borderColor={selected && onSelect ? 'brand.600' : 'white'}
       onClick={() => onSelect && onSelect()}
       mx={isMobile ? -3 : undefined}
+      cursor={isPharmacySelection ? 'pointer' : undefined}
```
Comment (samkotlove) [inline apps/patient/src/components/PharmacyCard.tsx:124]: what's this doing?

**Review by samkotlove — DISMISSED** [review]
Not quite sure about the logic. You're conditionally showing filters if there are open and 24hr pharmacies loaded. Why not just search for those pharmacies directly and always show the filters? The backend should support that

File: apps/patient/src/components/PharmacyCard.tsx (hunk 1 above)
Comment (jknapp25) [reply apps/patient/src/components/PharmacyCard.tsx]: better name would have been `selectable`, updated, on status page pharmacy card can't be selected so not using pointer

File: apps/patient/src/components/PharmacyFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}: PharmacyFiltersProps) => {
+  return (
+    <HStack>
+      <Text>Filter by</Text>
+      <Button
+        size="sm"
+        variant="filter"
+        isActive={enableOpenNow}
+        onClick={() => {
+          if (enable24Hr) setEnable24Hr(false);
```
Comment (jknapp25) [inline apps/patient/src/components/PharmacyFilters.tsx:23]: Filters are mutually exclusive since openNow and open24Hr is kind of weird

Comment (marisbest2) [reply apps/patient/src/components/PharmacyFilters.tsx:23]: I think it's fine to allow clicking both and I think it'd be a bit weird if clicking "openNow" turned off `24hr` filter

Comment (marisbest2) [reply apps/patient/src/components/PharmacyFilters.tsx:23]: Either way, you can do `setEnable24Hr(false)` without having the `if` statement

**Review by marisbest2 — APPROVED** [review]
Next year, we should do something about caching ratings results

File: apps/patient/src/api/external.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ export const getPlace = async (
     }
     return null;
   } catch (e) {
-    console.error(e);
+    console.log(e);
```
[the remark this answers is not in the document]
Comment (jknapp25) [reply apps/patient/src/api/external.ts:60]: It's just an error saying 0 results found so not that helpful to see it in datadog

**Comment by jknapp25** [comment]
@marisbest2 feedback from jomi is to go with mutually exclusive filters since open24hr includes open now

**Pull request merged**